### ISSUE TITLE
Stopped working after VS code and iOS latest updates

### ISSUE THREAD

**Issue by RahulPrabhune** [axilleasiv/vscode-javascript-repl-docs#15]
Hi Team,

Thank you for this extension, very useful.

Stopped working after VS code and iOS latest updates. No results for console.log('hello')

JS Repl:Run does nothing and hitting it the second time gives a warning: There is already a repl instance associated with this file

JS Repl: Stop for all files gives an error: Command 'JS Repl: Stop for all files' resulted in an error (IPC channel is already disconnected)

Please help

Thx
Rahul

**Comment by axilleasiv** [comment]
Hello, @RahulPrabhune thank you so much for your feedback. I am testing this in VS code 1.45.0 and macOS Catalina 10.15.3 and it seems to work fine. 

Can you share with me the version of the VS code and the macOS version that you have this issue?

**Comment by RahulPrabhune** [comment]
Thanks a lot.

macOS High Sierra 10.13.6 and Visual Studio Code 1.45.0

**Comment by RahulPrabhune** [comment]
Here is a video of the issue
https://www.youtube.com/watch?v=Ec6C6WO1tLM

Please advise a workaround.

**Comment by axilleasiv** [comment]
Hey @RahulPrabhune, thanks for recording the issue. It seems that the extension has an exception when you are trying to 'run the repl' for the `JavascriptPlayground.js` file. 

I am not sure what is happening here, but from what I have seen in your recorded video, there is an issue. It will be very helpful if you can share the `JavascriptPlayground.js` file with me, in order to debug the issue and find a fix for it.

**Comment by RahulPrabhune** [comment]
https://github.com/RahulPrabhune/JSPlayground/blob/master/JavascriptPlayground.js

Thanks
Rahul

**Comment by axilleasiv** [comment]
Hello, @RahulPrabhune thanks for sharing your code. I was thinking that maybe the problem was an infinite loop but I was not sure. After I checked your code now I am sure that this is the problem. 

What I mean is that the following function https://github.com/RahulPrabhune/JSPlayground/blob/master/JavascriptPlayground.js#L488
has an infinite while loop that keeps running forever until the process has no other memory. If you include your javascript file as a script in an HTML file and try to open this file with a browser you will see that the browser tab will crash or freeze or not responding.

The same is happening, if you will try to run this file or just the code that has the problem as a separate file in node.js 
```
function powerRanger(power, min, max) {
  let powValue = 0;
  let counter = min;
  let listArr = [];
  while (powValue < max) {
    powValue = Math.pow(power, 1 / counter);

    listArr.push(counter);
    counter++;
  }
  return listArr;
}
console.log(powerRanger(2, 49, 65));
```
You will get an error like the following `FATAL ERROR: CALL_AND_RETRY_LAST Allocation failed - JavaScript heap out of memory`

So if you just remove the console.log at the of the file https://github.com/RahulPrabhune/JSPlayground/blob/master/JavascriptPlayground.js#L496 or try to fix the while loop not to run forever, after that, if you try to 'run the repl' again, the extension will work again.

Finally, about this issue, I have opened the [following issue](https://github.com/axilleasiv/vscode-javascript-repl-docs/issues/13) some time ago in order for the extension to provide as a new feature a warning message and a handler to stop the current repl session when an infinite loop will happen. But it is not supported yet, maybe to a next version :)
